plugin/gdal/gdalvrt/src/test/java/it/geosolutions/imageio/plugins/vrt/ErdasImgVrtTest.java

```diff
@@ -69,4 +78,53 @@ public void jaiOperations() throws FileNotFoundException, IOException {
         ImageIOUtilities.disposeImage(image);
     }
 
+    @Test
+    public void bandSelection() throws FileNotFoundException, IOException {
+        if (!isGDALAvailable) {
+            return;
+        }
+        File file = TestData.file(this, fileName);
+
+        // ////////////////////////////////////////////////////////////////
+        // preparing to read
+        // ////////////////////////////////////////////////////////////////
+        final ParameterBlockJAI pbjImageRead;
+        pbjImageRead = new ParameterBlockJAI("ImageRead");
+        pbjImageRead.setParameter("Input", file);
+        ImageReadParam param = new ImageReadParam();
+        param.setSourceBands(new int[] { 2 });
+
+        SampleModel originalSampleModel = new ComponentSampleModel(
+                DataBuffer.TYPE_FLOAT, 256, 256, 3, 256 * 3, new int[] {0,1,2});
+        SampleModel sm = new ComponentSampleModel(
+                DataBuffer.TYPE_FLOAT, 256, 256, 1, 256, new int[] {0});
+
+        ColorModel cm = GDALUtilities.extractColorModel(null, originalSampleModel, 1);
+
+        ImageLayout layout = new ImageLayout().setTileHeight(256).setTileWidth(256);
+        layout.setColorModel(cm);
+        layout.setSampleModel(sm);
+        RenderingHints hints = new RenderingHints(JAI.KEY_IMAGE_LAYOUT, layout);
```

Comment:
Is it necessary to setup destination sample and color model? shouldn't it be able to guess it?

Reply:
Yes, is it needed.
The JAI ImageReadOpImage has a LayoutHelper that setup an ImageLayout when cm and sm are missing, taking the rawType from the imageReader:

<img width="1265" height="320" alt="image" src="https://github.com/user-attachments/assets/952710a4-5870-4348-b53d-5b527170c012" />

Resulting in setting up a wrong CM,SM combination :(